### ISSUE TITLE
Can't validate YAML files with tagged nodes (Ansible Vault)

### ISSUE THREAD

**Issue by mss** [hmcmanus/yaml-validator-hook#25]
I am merely a user of a BitBucket instance with this hook available. I am trying to enable it for our Ansible inventory repository and it breaks.

We use [variable-level encryption](https://docs.ansible.com/ansible/2.9/user_guide/vault.html#variable-level-encryption) for secrets. This will result in an error like this when I try to push to a repository with this hook enabled:
```
remote: ERROR: Invalid yaml file: XX01/test/inventory/group_vars/all/db.yml
remote: Can't construct a java object for !vault; exception=Invalid tag: !vault
remote:  in 'reader', line 15, column 20:
remote:     xxxxx_app_db_pass: !vault |
remote:                        ^
remote: 
To ssh://git.example.com/xxxx/cmdb_ansible.git
 ! [remote rejected]     master -> master (pre-receive hook declined)
error: failed to push some refs to 'ssh://git.example.com/xxxx/cmdb_ansible.git'
```

I don't know if this is issue fixable with the used YAML library in a generic way at all: It would be great if tagged nodes were supported with no further validation than YAML syntax validation.

**Comment by hmcmanus** [comment]
Just having a look at this now. Looks like it could be possible with a custom Constructor and Construct. I haven't had the time to look into this apart from a bit of research. I'm going to release an updated version without this for now but just putting a comment here to remind me when I get time to get back to it:

https://github.com/FasterXML/jackson-dataformats-text/issues/214#issue-674197227

Example of custom object:
https://github.com/Thinkofname/snakeyaml/blob/master/src/test/java/examples/SelectiveConstructorTest.java

**Comment by mss** [comment]
Thanks for getting back to this. And good timing: We planned to enable the plugin on all our repos and I totally forgot that I ran into this issue when I tried it first time back then.

**Comment by mss** [comment]
Based on your pointers I digged a bit as well and I think the linked example is not exactly what is needed since one would have to add every possible tag to the constructor and what I think is needed is something which just ignores the tags (ie. only checks for wellformedness).

But there is another example right next to it which looks good:

https://bitbucket.org/snakeyaml/snakeyaml/src/snakeyaml-1.33/src/test/java/examples/IgnoreTagsExampleTest.java

(I found this via the discussion at https://bitbucket.org/snakeyaml/snakeyaml/issues/376/how-to-ignore-unknown-local-tags which pointed at the old issue https://github.com/Thinkofname/snakeyaml/issues/39).

Since it was an idle Friday and I was in the mood of tinkering around I am having a shot at it...
